src/domain/organization/utils.ts

```diff
@@ -328,6 +335,35 @@ export const checkIsEditActionAllowed = ({
   return { editable: !warning, warning };
 };
 
+export const getEditMerchantWarning = ({
+  t,
+  user,
+}: {
+  t: TFunction;
+  user?: UserFieldsFragment;
+}): string => {
+  if (!user?.isSuperuser) {
```

Comment:
Financial admins should also be allowed to edit merchants.

Reply:
This was kind of intentional decision. The permission check are not working optimally in the case that user is financial admin and admin in the same organization . Financial admin user can only patch `web_store_merchants` values and admin user can update all other fields. But in the case that user is financial admin and admin the user has only financial admin permissions. So the reason for this was to avoid situation that admin user cannot edit the organization. 

But maybe better solution would be to allow admin + financial admin user to edit all organization fields in API and apply same permissions to the UI side. In that case we could:
- use PATCH organization with only `web_store_merchants` in the payload if user is financial admin
- use PUT organization without `web_store_merchants` in the payload if user is admin
- use PUT organization with full payload if user is admin and financial admin or superuser

Reply:
I think it can be considered a bug in the API that a user with financial admin + admin permissions is not able to do a PUT with full payload. I'll do a fix to allow that kind of a request. Otherwise the use of the different requests is what I too had in mind with the different user permissions when implementing the API changes so I think we can go with that solution 👍

Reply:
This is fixed in the most recent commit